docs/docs/tooling/profiler.md

```diff
@@ -76,60 +92,146 @@ unconstrained fn zero_out_array(ptr: u32, mut array: [u32; 32]) -> [u32; 32] {
     array
 }
 ```
-We chose to instead write our array inside of the unconstrained function. Then inside of our circuit we assert on every value in the array returned from the unconstrained function.
 
-This new program produces the following ACIR opcodes flamegraph with a total of 284 opcodes:
+Instead of writing our array in a fully constrained manner, we first write our array inside an unconstrained function and then assert every value in the array returned from the unconstrained function in a constrained manner.
```

Comment:
```suggestion
Instead of writing our array in a fully constrained context, we first write our array inside an unconstrained function. Then, we assert every value in the array returned from the unconstrained function in a constrained context.
```

Reply:
This sentence isn't a run-on, but I think it is a bit cleaner split up.